Mathlib/MeasureTheory/Integral/IntervalIntegral.lean

```diff
@@ -373,6 +373,74 @@ theorem Antitone.intervalIntegrable {u : ℝ → E} {a b : ℝ} (hu : Antitone u
 
 end
 
+/-!
+## Interval integrability of functions with even or odd parity
+-/
+section
+
+variable {f : ℝ → E}
+
+/-- An even function is interval integrable (with respect to the volume measure) on every interval
+of the form `0..x` if it is interval integrable (with respect to the volume measure) on every
+interval of the form `0..x`, for positive `x`.
+
+See `intervalIntegrable_of_even` for a stronger result.-/
+lemma intervalIntegrable_of_even₀ (h₁f : ∀ x, f x = f (-x))
+    (h₂f : ∀ x, 0 < x → IntervalIntegrable f volume 0 x) (t : ℝ) :
+    IntervalIntegrable f volume 0 t := by
+  rcases lt_trichotomy t 0 with h | h | h
+  · rw [IntervalIntegrable.iff_comp_neg]
+    conv => arg 1; intro t; rw [← h₁f]
+    simp [h₂f (-t) (by norm_num [h])]
+  · rw [h]
+  · exact h₂f t h
+
+/-- An even function is interval integrable (with respect to the volume measure) on every interval
+if it is interval integrable (with respect to the volume measure) on every interval of the form
+`0..x`, for positive `x`. -/
+theorem intervalIntegrable_of_even
+  (h₁f : ∀ x, f x = f (-x))
+  (h₂f : ∀ x, 0 < x → IntervalIntegrable f volume 0 x) :
+  ∀ x y, IntervalIntegrable f volume x y :=
```

Comment:
Mathlib has a rule "hypotheses left of colon", i.e. one would often prefer `(x y : \R) : IntervalIntegrable f volume x y` instead. Have you thought about this rule here? (Perhaps it's more useful to not follow it... but it would be nice to at least think about it.)

Reply:
I was not aware of this rule. I changed the formulation following your suggestion here and elsewhere.

Reply:
It's in the mathlib style guide: https://leanprover-community.github.io/contribute/style.html
If you've never seen it, it's worth taking a look.